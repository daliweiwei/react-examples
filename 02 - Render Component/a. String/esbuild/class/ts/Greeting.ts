/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

import { Component } from 'react';

export default class Greeting extends Component {
  render(): string {
    return 'Hello World!';
  }
}
