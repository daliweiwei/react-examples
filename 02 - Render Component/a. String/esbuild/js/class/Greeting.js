/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

import { Component } from 'react';

class Greeting extends Component {
  render() {
    return 'Hello World!';
  }
}

export default Greeting;
