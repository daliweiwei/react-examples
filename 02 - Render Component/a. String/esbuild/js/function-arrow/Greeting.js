/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

const Greeting = () => 'Hello World!';

export default Greeting;
