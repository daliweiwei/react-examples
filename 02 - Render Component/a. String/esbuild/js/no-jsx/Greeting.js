/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

function Greeting() {
  return 'Hello World!';
}

export default Greeting;
