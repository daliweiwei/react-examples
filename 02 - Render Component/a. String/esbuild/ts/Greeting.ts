/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

export default function Greeting() {
  return 'Hello World!';
}
