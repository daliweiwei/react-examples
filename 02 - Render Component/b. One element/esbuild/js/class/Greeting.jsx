/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

import { Component } from 'react';

class Greeting extends Component {
  render() {
    return <h1>Hello World!</h1>;
  }
}

export default Greeting;
