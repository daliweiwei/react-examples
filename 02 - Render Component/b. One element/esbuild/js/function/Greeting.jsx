/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

function Greeting() {
  return <h1>Hello World!</h1>;
}

export default Greeting;
