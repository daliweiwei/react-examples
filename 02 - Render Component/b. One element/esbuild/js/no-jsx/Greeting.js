/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

import React from 'react';

function Greeting() {
  return React.createElement('h1', null, 'Hello World!');
}

export default Greeting;
