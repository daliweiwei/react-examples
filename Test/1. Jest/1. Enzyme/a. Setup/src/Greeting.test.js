import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Greeting from './Greeting';

configure({ adapter: new Adapter() });

/// Enzyme ///

describe('Greeting', () => {
  it('renders hello world snapshot', () => {
    const component = shallow(<Greeting />);
    let tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
