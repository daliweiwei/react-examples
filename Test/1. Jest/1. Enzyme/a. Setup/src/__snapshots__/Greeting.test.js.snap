// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting renders hello world snapshot 1`] = `
<h1
  className="greeting"
>
  Hello World!
</h1>
`;
